src/sampleutil.cpp

```diff
@@ -296,26 +296,31 @@ void SampleUtil::convertS16ToFloat32(CSAMPLE* _RESTRICT pDest, const SAMPLE* _RE
 }
 
 // static
-bool SampleUtil::sumAbsPerChannel(CSAMPLE* pfAbsL, CSAMPLE* pfAbsR,
+SampleUtil::CLIPPING SampleUtil::sumAbsPerChannel(CSAMPLE* pfAbsL, CSAMPLE* pfAbsR,
         const CSAMPLE* pBuffer, int iNumSamples) {
     CSAMPLE fAbsL = CSAMPLE_ZERO;
     CSAMPLE fAbsR = CSAMPLE_ZERO;
-    CSAMPLE clipped = 0;
+    CSAMPLE clippedL = 0;
+    CSAMPLE clippedR = 0;
```

Comment:
I build Mixxx as usual. In the console output I search for the g++ command line of the file of interest. 
I use the found command line and add -ftree-vectorizer-verbose=2. 
All loops that are vectorized are printed to console, identified by its line number.

Some of this is described at the top of sampleutil.cpp
 
> Can we just leave the current solution?
> return static_cast((clippedL>0) | ((clippedR>0)<<1));

Yes, it is your choice. I would pick the later one with the | operator override because this 
avoids to introduce the constant "1" for shifting. 
Just my two cents. 

By the way: the Windows SDK features DEFINE_ENUM_FLAG_OPERATORS
which adds all flag operators for you. So it seams to be the usual way to avoid all the casting.